apps/sms/lib/bridge/plugins/dispose/service.js

```diff
@@ -0,0 +1,40 @@
+!((define)=>{define((require,exports,module)=>{
+
+  /**
+   * Mini Logger
+   *
+   * @type {Function}
+   */
+
+  var debug = 1 ? function(arg1, ...args) {
+    var type = `[${self.constructor.name}][${location.pathname}]`;
+    console.log(`[DisposableService]${type} - "${arg1}"`, ...args);
+  } : () => {};
+
+  module.exports = function(service) {
+    debug('attaching plugin');
+
+    service.method('@@requestDisposal', (clientId) => {
+      debug('Client %s is requesting to dispose service', clientId);
+
+      let canBeDisposed = true;
+      service.eachClient((client) => {
+        if (client.id !== clientId) {
+          canBeDisposed = false;
+        }
+      });
+
+      if (canBeDisposed) {
+        // Use setTimeout to allow service to respond.
+        setTimeout(() => service.emit('dispose'), 0);
```

Comment:
I'd do both: emit the event in case it's useful but still close it right after this (but maybe then there could be issues where the event is not sent ?)